### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by grantdwoods**

### CODE REVIEW

**Comment by grantdwoods** [comment]
If anyone understands the GroupService well, I would appreciate feedback on how I am trying to add/remove a user from a group! Will write unit tests for this later today.

File: SecretSanta/test/SecretSanta.Domain.Tests/Services/UserServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        public void CloseConnection()
+        {
+            SqliteConnection.Close();
+        }
+        [TestMethod]
+        public void AddUser()
+        {
+            UserService userService;
+
+            User user = new User
```
Comment (ConnerV42) [inline SecretSanta/test/SecretSanta.Domain.Tests/Services/UserServiceTest.cs]: You could add a CreateUser class, in order to reduce code duplication in this class, similar to your Gift class'es CreateGift helper method.

File: SecretSanta/src/SecretSanta.Domain/Services/GroupService.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+        public Group AddGroup(Group group)
+        {
+            Context.Groups.Add(group);
+            Context.SaveChanges();
+
+            return group;
+        }
+
+        public void AddUserToGroup(int groupId, UserGroup userGroup)
```
Comment (ConnerV42) [inline SecretSanta/src/SecretSanta.Domain/Services/GroupService.cs]: This looks like a good implementation. My only suggestion would be to possibly increase the encapsulation of this method by taking in the Entire Group object and then just grabbing the group Id from within the method.

File: SecretSanta/src/SecretSanta.Domain/Interface/IEntity.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SecretSanta.Domain.Interface
```
[the remark this answers is not in the document]
Comment (grantdwoods) [reply SecretSanta/src/SecretSanta.Domain/Interface/IEntity.cs]: Good point, I will consider if there is a compelling reason to keep this structure around.

File: SecretSanta/test/SecretSanta.Domain.Tests/Services/UserServiceTest.cs (hunk 1 above)
Comment (grantdwoods) [reply SecretSanta/test/SecretSanta.Domain.Tests/Services/UserServiceTest.cs]: That's a good point, I also forgot about C#'s default parameter option until I started looking at everyone's code.  I will definitely refactor this!

**Comment by grantdwoods** [comment]
Having a hard time being able to filter out Users based on their collection of UserGroups. The last test in the GroupServiceTest is "passing" because there is nothing verifying the method works for now...

**Pull request closed without merging**